doc/userguide/src/CreatingTestData/AdvancedFeatures.rst

```diff
@@ -399,6 +399,103 @@ lower limit, upper limit and step.
+inside the list.  Robot Framework has a special
+`FOR index ... IN ENUMERATE ...` syntax for this situation.
+This syntax is derived from the
+`Python built-in function <https://docs.python.org/2/library/functions.html#enumerate>`.
+
+:name:`:FOR` :name:`IN ENUMERATE` loops work just like regular :name:`:FOR` loops,
+except the cell after its loop variables must say :name:`IN ENUMERATE`,
+and they must have an additional index variable before any other loop-variables.
+That index variable has a value of `0` for the first loop, `1` for the second, etc.
+
+For example, the following two test cases do the same thing:
+
+
+.. table:: For in enumerate comparison
+   :class: example
+   ================= ================ =============== ============= =================== =======
+   Test Case         Action           Argument        Argument      arg                 arg
+   ================= ================ =============== ============= =================== =======
+   Manual Enumerate  [Documentation]  How to do this  the old way
+   \                 @{list}=         Create List     a             b                   c
+   \                 ${index}=        Set Variable    ${-1}
+   \                 :FOR             ${item}         IN            @{list}
+   \                                  ${index}=       Set Variable  ${index} + 1
+   \                                  My Keyword      ${index}      ${item}
+   For In Enumerate  [Documentation]  How to do this  with          `FOR IN ENUMERATE`
+   \                 @{list}=         Create List     a             b                   c
+   \                 :FOR             ${index}        ${item}       IN ENUMERAGE        @{list}
+   \                                  My Keyword      ${index}      ${item}
+   ================= ================ =============== ============= =================== =======
+
+Just like with regular For-loops, you can loop over multiple values per loop iteration
+as long as the number of values in your list evenly is divisible by the number of
+loop-variables (excluding the first, `index` variable).
+
+.. table:: For in enumerate examples
+   :class: example
+   ================= ================ ============ ========= ========= ============= =======
+   Test Case         Action           Argument     Argument  arg       arg
+   ================= ================ ============ ========= ========= ============= =======
+   For In Enumerate  [Documentation]  Two values   at once
+   \                 @{list}=         Create List  a         b         c             d
+   \                 :FOR             ${index}     ${item}   ${thing}  IN ENUMERAGE  @{list}
+   \                                  Log Many     ${index}  ${item}   $[thing}
+   ================= ================ ============ ========= ========= ============= =======
+
+The above test case will log `0,a,b` and `1,c,d`.
+
+For In Enumerate loops are new in Robot 2.9.
```

Comment:
I fixed all instances of this, except for one spot in a table where I used "Version 2.9" to prevent widening a column excessively.